### PR TITLE
test: Update safari-ssl-e2e tests for WebDriverIO compatibility

### PR DESCRIPTION
**Pull request by mwakizaka**
Ref: https://github.com/appium/appium/issues/16403

### CODE REVIEW

**Comment by mwakizaka** [comment]
> 2023-02-25T04:02:58.3277480Z uncaughtException: invalid rimraf options
2023-02-25T04:02:58.3330020Z Error: invalid rimraf options

it seems like the error happened around https://github.com/appium/node-simctl/blob/75a7e4847bb4c4e572257a49514e4df9b6e3f921/lib/subcommands/keychain.js#L23 for some reasons..

**Comment by mwakizaka** [comment]
^ broken most probably since https://github.com/appium/node-simctl/commit/93bce628ff65f53695f1c91a0cc13ed01428c3b9

**Comment by mykola-mokhnach** [comment]
Thank you @mwakizaka for reporting these issues. I hope https://github.com/appium/node-simctl/pull/174 will address them

**Review by mykola-mokhnach — APPROVED** [review]

File: test/functional/web/safari-ssl-e2e-specs.js [hunk 1]
```diff
@@ -1,125 +1,140 @@
...
+chai.use(chaiAsPromised);
+
+const HTTPS_PORT = 9762;
+
+const LOCAL_HTTPS_URL = `https://localhost:${HTTPS_PORT}/`;
+
+let caps;
+let pemCertificate;
+
+if (!process.env.REAL_DEVICE && !process.env.CLOUD) {
```
Comment (mykola-mokhnach) [inline test/functional/web/safari-ssl-e2e-specs.js:25]: shall we deprecate these env vars in this PR?

Comment (mwakizaka) [reply test/functional/web/safari-ssl-e2e-specs.js]: done as https://github.com/appium/appium-xcuitest-driver/pull/1507/commits/b7ef6778c0d592bcb8d909c0a2e43aa140f06684 and https://github.com/appium/appium-xcuitest-driver/pull/1507/commits/4ae2757ff59d8b985a0c0115c98c48eec5ad4afc

**Pull request merged**